src/geop/processing/src/main/scala/com/azavea/mmw/TR55.scala

```diff
@@ -0,0 +1,395 @@
+    2479808 -> "C/D",
+    2479809 -> "B",
+    2479810 -> "B",
+    2479811 -> "C/D",
+    2479812 -> "D",
+    2479813 -> "D",
+    2479814 -> "D",
+    2479815 -> "D",
+    2479816 -> "D",
+    2479817 -> "C",
+    2479818 -> "C",
+    2479819 -> "B",
+    2479820 -> "B",
+    2479821 -> "B",
+    2479822 -> "B",
+    2479835 -> "C/D",
+    2479836 -> "C/D",
+    2479837 -> "C",
+    2479838 -> "C",
+    2479839 -> "C",
+    2479840 -> "C",
+    2479841 -> "C/D",
+    2479842 -> "C",
+    2479843 -> "C",
+    2479844 -> "C",
+    2479845 -> "C",
+    2479846 -> "C/D",
+    2479847 -> "C",
+    2479848 -> "C",
+    2479849 -> "C",
+    2479850 -> "C",
+    2479851 -> "C",
+    2479852 -> "C",
+    2479853 -> "B/D",
+    2479854 -> "B/D",
+    2479855 -> "C",
+    2479856 -> "B",
+    2479857 -> "B",
+    2479858 -> "B",
+    2479859 -> "C",
+    2479860 -> "C",
+    2479861 -> "C",
+    2479862 -> "B",
+    2479865 -> "B/D",
+    2479866 -> "D",
+    2479867 -> "D",
+    2479868 -> "D",
+    2479869 -> "B",
+    2479870 -> "B",
+    2479871 -> "C/D",
+    2479872 -> "C/D",
+    2479873 -> "B",
+    2479874 -> "B",
+    2479875 -> "B",
+    2479876 -> "B",
+    2479877 -> "B",
+    2479878 -> "C/D",
+    2479879 -> "B",
+    2479880 -> "B",
+    2479881 -> "B",
+    2479882 -> "C",
+    2479883 -> "C",
+    2479884 -> "C",
+    2479885 -> "B",
+    2479886 -> "B",
+    2479887 -> "B",
+    2479888 -> "B",
+    2479889 -> "B",
+    2479890 -> "B",
+    2479891 -> "B",
+    2479892 -> "B",
+    2479893 -> "B",
+    2479894 -> "B",
+    2479895 -> "B",
+    2479896 -> "B",
+    2479897 -> "C/D",
+    2479898 -> "C/D",
+    2479899 -> "C/D",
+    2479900 -> "B",
+    2479901 -> "B")
+
+  /** Returns a map of values to counts */
+  def combinations(nlcd: RasterRDD[SpatialKey], soil: RasterRDD[SpatialKey], polygon: Polygon): Map[(Int, Option[String]), Int] = {
+    val mapTransform = nlcd.metaData.mapTransform
+    val nlcdAndSoil = nlcd.join(soil)
+    val histogram = nlcdAndSoil
+      .map { case (key, (nlcdTile, soilTile)) =>
+        val extent = mapTransform(key) // transform spatial key to map extent
+        val rasterExtent = RasterExtent(extent, nlcdTile.cols, nlcdTile.rows) // transform extent to raster extent
+        val clippedPolygon = polygon & extent
+        val localHistogram = mutable.Map[(Int, Option[String]), Int]()
+
+        clippedPolygon match {
+          case PolygonResult(p) =>
+            Rasterizer.foreachCellByPolygon(p, rasterExtent)(
+              new Callback {
+                def apply(col: Int, row: Int): Unit = {
+                  val soilType = wordToSoil.get(soilTile.get(col,row))
+                  if (soilType != None) {
+                    val c = (nlcdTile.get(col, row), soilType)
+                    if(!localHistogram.contains(c)) { localHistogram(c) = 0 }
+                    localHistogram(c) += 1
+                  }
+                }
+              }
+            )
+          case unexpected =>
```

Comment:
Should handle all the cases, without a catchall. You can use the `|` operator to catch a set of them:

```scala
clippedPolygon match {
  case PolygonResult(p) => //something
  case MultiPolygonResult(mp) | LineResult(l) => // something else
  case NoResult => // do nothing   
```